fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/util/JDBCQueryBuilder.java

```diff
@@ -1124,6 +1197,79 @@ private SqlQueryData processQuantityParm(Class<?> resourceType, Parameter queryP
         log.exiting(CLASSNAME, METHODNAME, whereClauseSegment.toString());
         return queryData;
     }
+    
+    @Override
+    protected SqlQueryData processUriParm(QueryParameter queryParm) throws FHIRPersistenceException {
+        return processUriParm(queryParm, PARAMETERS_TABLE_ALIAS);
+    }
+
+    /**
+     * Creates a query segment for a URI type parameter.
+     * @param queryParm - The query parameter
+     * @param tableAlias - An alias for the table to query
+     * @return SqlQueryData - An object containing query segment
+     * @throws FHIRPersistenceException
+     */
+    protected SqlQueryData processUriParm(QueryParameter queryParm, String tableAlias) throws FHIRPersistenceException {
+        final String METHODNAME = "processUriParm";
+        log.entering(CLASSNAME, METHODNAME, queryParm.toString());
+        SqlQueryData parmRoot = this.processStringParm(queryParm, tableAlias);
+        log.exiting(CLASSNAME, METHODNAME, parmRoot.toString());
+        return parmRoot;
+    }
+
+    @Override
+    protected SqlQueryData processCompositeParm(Class<?> resourceType, QueryParameter queryParm) throws FHIRPersistenceException {
+        return processCompositeParm(resourceType, queryParm, PARAMETERS_TABLE_ALIAS);
+    }
+
+    /**
+     * Creates a query segment for a URI type parameter.
+     * @param queryParm - The query parameter
+     * @param tableAlias - An alias for the table to query
+     * @return SqlQueryData - An object containing query segment
+     * @throws FHIRPersistenceException
+     */
+    private SqlQueryData processCompositeParm(Class<?> resourceType, QueryParameter queryParm, String tableAlias) throws FHIRPersistenceException {
+        final String METHODNAME = "processStringParm";
+        log.entering(CLASSNAME, METHODNAME, queryParm.toString());
+
+        StringBuilder whereClauseSegment = new StringBuilder();
+        SqlQueryData queryData;
+        List<Object> bindVariables = new ArrayList<>();
+
+        // Build this piece of the segment:
+        // (P1.PARAMETER_NAME_ID = x AND
+        populateNameIdSubSegment(whereClauseSegment, queryParm.getCode(), tableAlias);
+
+        whereClauseSegment.append(AND).append(LEFT_PAREN);
+        
+        String valueSeparator = "";
+        for (QueryParameterValue compositeValue : queryParm.getValues()) {
+            List<QueryParameter> components = compositeValue.getComponent();
+            
+            whereClauseSegment.append(valueSeparator);
+            
+            String componentSeparator = "";
+            for (int j = 0; j < components.size(); j++) {
+                try {
+                    QueryParameter component = components.get(j);
+                    SqlQueryData subQueryData = buildQueryParm(resourceType, component, tableAlias + "_p" + (j+1));
+                    whereClauseSegment.append(componentSeparator + subQueryData.getQueryString());
```

Comment:
does componentSeparator ever change? 
also split into separate append operations.

Reply:
sorry... down the stream it's set.